### ISSUE TITLE
Can't update to new version (via update button)

### ISSUE THREAD

**Issue by eL3cTr1c4L** [Maassoft/ColorControl#204]
Hi, I'm having this issue where the update does not work, please take a look at the screengrab:

![Screenshot 2023-04-19 155309](https://user-images.githubusercontent.com/131279564/233114855-0681937c-9432-4d6c-8440-d08f3196ce45.png)

I presume it's because of MS Defender folder security, but I'm not sure.

LE: added the LOGS as well:
![image](https://user-images.githubusercontent.com/131279564/233116980-0f22a445-4743-4e89-a391-264b16781296.png)

Thanks,
Robert

**Comment by Maassoft** [comment]
Thanks for the report.
That's strange, but it could indeed be Microsoft Defender that's causing this. Is there anything in the log of Defender itself?
I don't know if there's a simple workaround for this, other than excluding the folder from Defender, but that may be bad for security. Another way is that I have to get a certificate for the app.

**Comment by javsp** [comment]
Same error here when updating from 9.1.0.0 to 9.1.1.0.

Disabling Microsoft Defender causes the same error. Should I update the app manually or you want I try anything else?

**Comment by Maassoft** [comment]
Can you check if there's a file "update.zip" in this folder:
`C:\Windows\System32\config\systemprofile\AppData\Roaming\Maassoft\ColorControl`
You probably need admin rights to view that folder.
If the file exists, can you manually extract it?

**Comment by javsp** [comment]
> Can you check if there's a file "update.zip" in this folder: `C:\Windows\System32\config\systemprofile\AppData\Roaming\Maassoft\ColorControl` You probably need admin rights to view that folder. If the file exists, can you manually extract it?

Yep, the file is there (9276 KB). Can I delete it once I extract it?

**Comment by eL3cTr1c4L** [comment]
Same here:

![image](https://user-images.githubusercontent.com/131279564/233852385-defd1511-d0c0-427a-9621-7eb130a82e0f.png)

**Comment by Maassoft** [comment]
That's weird. It's almost as if the zip-file is compressed using an incompatible compression method.
If you copy the .zip to a "normal" folder, can you extract it then?
Another thing to check: open the properties of the file, maybe it is blocked by security reasons (because downloaded from the internet).

**Comment by javsp** [comment]
These are the only files I can extract from update.zip (WinRAR) before I get the error:

![Sin título](https://user-images.githubusercontent.com/33856667/233868395-4823637c-6ece-4526-ad45-d2e5a2617403.png)

I've tried to unzip it moving the file to the Desktop.

I don't see any blocked permission in properties.

**Comment by Maassoft** [comment]
I think I found a possible bug: if the file "update.zip" already existed, the downloaded contents of the new .zip is written into this file and could cause it to go corrupt. I think if you remove the old "update.zip" first from the systemprofile folder and then try to update it would probably work.

**Comment by javsp** [comment]
I've already done a manual update. On the next update I'll do what you say if I have the same error.

**Comment by javsp** [comment]
@Maassoft I got no errors this time. I updated it by pressing the Update button on the top right corner. By the way, I don't know if it matters but "Automatically check for updates" is enabled and "Automatically install updates" is disabled (as I prefer to know when an update is available to download).

**Comment by Maassoft** [comment]
No, that shouldn't matter. If updating to v9.3 also worked without errors we can close this issue.
